packages/harpguru-core/src/components/harp-face-fragment/hooks/use-harp-rows/use-harp-rows.tsx

```diff
@@ -0,0 +1,52 @@
+import { useGlobal } from 'reactn'
+import React, { ReactElement } from 'react'
+import { InteractionIds } from 'harpparts'
+
+import { mapRowToBlowDrawIds } from '../map-row-to-blow-draw-ids'
+import { HarpRow } from '../../../harp-row'
+import type { XRange } from '../../../../types'
+import { useViewableMatrices } from '../../../../hooks'
+
+type HarpRows = {
+  readonly top: ReactElement[]
+  readonly bottom: ReactElement[]
+}
+
+export const useHarpRows = (xRange: XRange): HarpRows => {
+  const { viewableInteractionMatrix } = useViewableMatrices()
+  const viewableBlowDrawIdsMap = viewableInteractionMatrix.map(
+    mapRowToBlowDrawIds
+  )
+
+  const drawIndex = viewableBlowDrawIdsMap.indexOf(InteractionIds.Draw)
+  const topRowsPrimer = viewableBlowDrawIdsMap.slice(0, drawIndex)
+  const bottomRowsPrimer = viewableBlowDrawIdsMap.slice(drawIndex)
+
+  // TODO: this is a bit clumsy. There are 2 options here:
```

Comment:
TODO here

Reply:
Raised as techdebt in https://github.com/js-jslog/harpguru/issues/113#issuecomment-922351622

The work done to make the app sizing dynamic (at #107) was used as a platform for additional speculative pull requests to ensure that this was the correct direction to start down.

Rather than addressing the finer details in this or subsequent pull request and potentially struggling to merge up to those subsequent branches I will prefer to record the issues in each merge request and resolve them afterwards.

This has a limit of course. Major issues will not be merged to master. I will need to find a way around that if it happens.